### PR TITLE
You can eat as fast as you can click, but eating too fast makes you choke

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
This idea has made me giggle for a while now, so I may as well make the PR since it's pretty simple.

https://cdn.discordapp.com/attachments/326831214667235328/708298898485018645/Space_Station_13_2020-05-08_08-45-46.mp4

There is no cooldown when eating food on yourself, but however much you bite out of the food is placed in your head to chew and swallow. If you eat too much food too quick, you can start choking. If you're lucky, you'll automatically vomit up the food blocking your throat, but otherwise you'll need someone to apply the Heimlich Maneuver by aggro grabbing you, and interacting with you on a non-grab intent. Failing to clear the blockage may lead to suffocation!

The voracious trait has also been changed to make you chew food 1.5x faster.

## Why It's Good For The Game
Adds mechanical depth to eating, raises the skill ceiling of not dying to a plate of chicken tenders, baits assistants into suffocating themselves for the vine

## Changelog
:cl: Ryll/Shaps
add: You can now eat food as fast as you can click!
add: Just because you CAN doesn't mean you should though!
tweak: The voracious perk now makes you chew food 50% faster.
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
Not a big fan of the spam click change, it's going to result in new players instantly choking on food without any clear indication that this is a thing, I'd prefer if this were behind a quirk, it'll also probably be changed once edible component is used for everything but aslong as thats not a thing it's also not really a factor i suppose.

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -655,3 +655,62 @@
...
+			else if(prob(50))
+				C.losebreath += 3
+				C.visible_message("<span class='danger'>[C]'s eyes seem to darken noticeably as [C.p_their()] flailing begins to die down!</span>", "<span class='userdanger'>CAN'T- BREATHE- IS THIS- IT?!</span>", "<span class='danger'>You hear the wheezing embodiment of despair!</span>")
+			else
+				C.losebreath += 4
+
+/obj/screen/alert/status_effect/choking
+	name = "Choking"
+	desc = "You're choking! Find someone to apply the Heimlich Maneuver!"
+	icon_state = "gross3"
```
Comment (Qustinnus) [inline code/datums/status_effects/debuffs.dm:716]: get your own icon you thief >:^(

File: code/modules/food_and_drinks/food/snacks.dm [hunk 2]
```diff
@@ -147,6 +146,11 @@ All foods are distributed among various categories. Use common sense.
 				var/fraction = min(bitesize / reagents.total_volume, 1)
 				reagents.trans_to(M, bitesize, transfered_by = user, method = INGEST)
 				bitecount++
+				if(iscarbon(M))
+					var/mob/living/carbon/C = M
+					var/obj/item/bodypart/head/hed = C.get_bodypart(BODY_ZONE_HEAD)
+					if(hed)
+						hed.mouthful += bitesize
```
Comment (Qustinnus) [inline code/modules/food_and_drinks/food/snacks.dm:153]: please don't add more kaka code to food code while it's still being moved to the component :(

Comment (Qustinnus) [reply code/modules/food_and_drinks/food/snacks.dm:153]: (or atleast add it to the component too)

File: code/modules/surgery/bodyparts/head.dm [hunk 3]
```diff
@@ -246,6 +255,60 @@
...
+			if(121 to 140)
+				if(prob(30))
+					to_chat(owner, "<span class='warning'>Your jaw is rebelling against your gluttony as it struggles to make room for more food!</span>")
+					this_chew *= 0.5
+				else if(prob(40))
+					owner.visible_message("<span class='danger'>[owner] appears to start tearing up as [owner.p_they()] continue[owner.p_s()] shoveling food in [owner.p_their()] mouth with horrifying gusto!</span>", "<span class='warning'>It's so hard to breathe! You're not sure how much more you can take!</span>")
+					this_chew *= 0.4
+					owner.losebreath += 4
+				else if(prob(70))
+					owner.visible_message("<span class='danger'>[owner] starts crying in agony, even as [owner.p_they()] continue[owner.p_s()] packing food into [owner.p_their()] mouth!</span>", "<span class='warning'>It hurts!! You think you might die!</span>")
```
Comment (XDTM) [inline code/modules/surgery/bodyparts/head.dm:302]: Won't these messages also show if the user's mouthful is above the choking threshold? They might not be actively eating anymore at that point

Comment (Ryll-Ryll) [reply code/modules/surgery/bodyparts/head.dm:302]: Getting all the food into your mouth is one thing, actually swallowing it is another deal entirely

**Comment by ATH1909** [comment]
> Not a big fan of the spam click change, it's going to result in new players instantly choking on food without any clear indication that this is a thing, I'd prefer if this were behind a quirk, it'll also probably be changed once edible component is used for everything but aslong as thats not a thing it's also not really a factor i suppose.

This.

Also, this is gonna make gene-modded apples and potatoes from botany a severe and unexpected choking hazard, since you always eat them in a single bite.

This is also gonna make eating in general more of a chore than it already is. This is, like, manually blinking/breathing levels of tedi- hey, that gives me an idea.

Why not just make this feature restricted to when admins enable that admin-only manual blinking/breathing mode?

**Comment by TheVekter** [comment]
Hooray! You throw up!

**Comment by SkeletalElite** [comment]
This adds nothing of value to the game.
It's confusing for new players.
It's just annoying for old players.
Clicking yourself with food to make hunger meter go up isn't something that needs to have a conscious effort.

**Comment by Ryll-Ryll** [comment]
This was clearly one of those sleep deprived ideas that only seems good if you don't think about it. Wish it well folks, it's time to lay it down

**Comment by wesoda25** [comment]
relatable

**Pull request closed without merging**